smoketest/directives.py

```diff
@@ -365,7 +365,11 @@ def _generate_directives_from_xml(self):
         else:
             try:
                 url = self.filename
-                response = urllib.request.urlopen(url)
+                response = ''
+                if hasattr(urllib, 'request'):
+                    response = urllib.request.urlopen(url)
```

Comment:
I think it might be useful to add a comment for which case applies to Python 2 vs. Python 3, should be helpful when we eventually drop support for Python 2 one day.